packages/smart-account/src/SmartAccount.ts

```diff
@@ -587,7 +587,7 @@ class SmartAccount extends EventEmitter {
     if (updateImplTrx.data != '0x') {
       transactions.unshift(updateImplTrx)
     }
-    const response = await aaSigner.sendTransactionBatch(transactions, false)
+    const response = await aaSigner.sendTransactionBatch(transactions, this)
```

Comment:
why false argument is changed to 'this'
wouldn't that mess up with the order?

Reply:
It was wrong initially, sendTransactionBatch function just take two arguments. One is transaction object and other is event emitter, initially false value was supplied that was wrong so have corrected that

Reply:
and sendTransaction takes three args now? without breaking ethers signer interface

Reply:
yes, here are the args

```
 transaction: TransactionRequest,
 walletDeployOnly = false,
 engine?: any // EventEmitter
```